### DISCUSSION TITLE
Another issue running TheengsExplorer

### DISCUSSION THREAD

**Discussion by NateEaton** [theengs/explorer#20] [Q&A] [answered]
With plans to try creating a decoder or two, I figured I would start by getting explorer working. I am working from versions of decoder  and explorer just forked today and cloned to the Linux environment on my Pixel Slate. I installed all the listed dependencies (though I didn't see a reference to Bleak as a dependency on the explorer README.md but when I got the error indicating I needed it, it made sense so I installed). I had a few issues with the build of decoder but they were mostly because I needed to upgrade pip and had missed submodule when cloning. 

However, when I try to get explorer to run, I get this:
`$ python3 TheengsExplorer/__init__.py
Traceback (most recent call last):
  File "/home/nateeatonjr/explorer/TheengsExplorer/__init__.py", line 26, in <module>
    from textual.widgets import Footer, ScrollView
  File "/home/nateeatonjr/.local/lib/python3.9/site-packages/textual/widgets/__init__.py", line 62, in __getattr__
    raise ImportError(f"Package 'textual.widgets' has no class '{widget_class}'")
ImportError: Package 'textual.widgets' has no class 'ScrollView'`

My guess is that I've missed a step or screwed up installing Textual but I've gone back over everything several times and no joy. 

Suggestions?

**Comment by koenvervloesem** [comment, marked as answer]
The Textual API has changed a lot recently, and I still need to rewrite Theengs Explorer. You need Textual 0.1.18 and bluetooth-numbers 0.2.1 to be able to run the current version.

**Comment by NateEaton** [comment]
Thanks for the update. I'll give that a try.

---

**Comment by NateEaton** [comment]
That did the trick, am able to execute now. Unfortunately, running it on a Raspberry Pi via an ssh terminal connection from my Pixel Slate only appears to work. Explorer starts up and is displaying all the expected information and initially the window will scroll if I try scrolling with mousepad on the Pixel Slate but the keyboard is unresponsive (no way to exit the app either with Q or Ctrl-C). Eventually it no longer appears to be running.  When I tried with terminal session on my Mac, similar results except if I hit Q the bottom bar disappears and is replaced by "=2". 

Although that pi is generally running headless, I can connect it to a monitor. I'll give that a try in the next few days.

**Comment by koenvervloesem** [comment]
If getting the Theengs Decoder output isn't important for you, you can always try [HumBLE Explorer](https://github.com/koenvervloesem/humble-explorer). This is a general-purpose BLE scanner I built with the new Textual API from the start. I noticed that it's much more responsive and less CPU-hungry than Theengs Explorer.

**Comment by castaway** [comment]
Hi, I just stumbled across the same issue.. but instead of finding this reply, I beat things with  a stick until they appeared to work on textual 0.9.1, would you like a patch?
